cloudcore crash when no nodeSelector provide

<!-- Please use this template while reporting a bug and provide as much info as possible. Thanks!-->
**What happened**:
when we create a device instance without nodeSelector, cloudcore will crash.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1363d8f]

goroutine 1416 [running]:
github.com/kubeedge/kubeedge/cloud/pkg/devicecontroller/controller.(*DownstreamController).deviceAdded(0xc000562380, 0xc00122e4e0)
        /go/src/github.com/kubeedge/kubeedge/cloud/pkg/devicecontroller/controller/downstream.go:365 +0x9f
github.com/kubeedge/kubeedge/cloud/pkg/devicecontroller/controller.(*DownstreamController).syncDevice(0xc000562380)
        /go/src/github.com/kubeedge/kubeedge/cloud/pkg/devicecontroller/controller/downstream.go:158 +0x24e
created by github.com/kubeedge/kubeedge/cloud/pkg/devicecontroller/controller.(*DownstreamController).Start
        /go/src/github.com/kubeedge/kubeedge/cloud/pkg/devicecontroller/controller/downstream.go:841 +0xd0
```
**What you expected to happen**:
cloudcore works properly on device without nodeselector, or just reject it.

**How to reproduce it (as minimally and precisely as possible)**:
```
[root@masternode1 api-test]# cat create-device-instance-test.yaml
apiVersion: devices.kubeedge.io/v1alpha1
kind: Device
metadata:
  name: sensor-tag-instance-01-test
  labels:
    description: TISimplelinkSensorTag
    manufacturer: TexasInstruments
    model: CC2650
spec:
  deviceModelRef:
    name: sensor-tag-model-test
```
```
kubectl create -f create-device-instance-test.yaml
```
**Anything else we need to know?**:
I think these codes cause the problem
```
cloud/pkg/devicecontroller/controller/downstream.go:172:        configMap, ok := dc.configMapManager.ConfigMap.Load(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0])
cloud/pkg/devicecontroller/controller/downstream.go:177:                nodeConfigMap.Name = DeviceProfileConfigPrefix + device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0]
cloud/pkg/devicecontroller/controller/downstream.go:183:                dc.configMapManager.ConfigMap.Store(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0], nodeConfigMap)
cloud/pkg/devicecontroller/controller/downstream.go:204:        dc.configMapManager.ConfigMap.Store(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0], nodeConfigMap)
cloud/pkg/devicecontroller/controller/downstream.go:365:        if len(device.Spec.NodeSelector.NodeSelectorTerms) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values) != 0 {
cloud/pkg/devicecontroller/controller/downstream.go:370:                resource, err := messagelayer.BuildResource(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0], "membership", "")
cloud/pkg/devicecontroller/controller/downstream.go:447:// isNodeSelectorUpdated checks if nodeSelector is updated
cloud/pkg/devicecontroller/controller/downstream.go:448:func isNodeSelectorUpdated(oldTwin *v1.NodeSelector, newTwin *v1.NodeSelector) bool {
cloud/pkg/devicecontroller/controller/downstream.go:449:        return !reflect.DeepEqual(oldTwin.NodeSelectorTerms, newTwin.NodeSelectorTerms)
cloud/pkg/devicecontroller/controller/downstream.go:469:        if len(device.Spec.NodeSelector.NodeSelectorTerms) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values) != 0 {
cloud/pkg/devicecontroller/controller/downstream.go:470:                configMap, ok := dc.configMapManager.ConfigMap.Load(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0])
cloud/pkg/devicecontroller/controller/downstream.go:550:                dc.configMapManager.ConfigMap.Store(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0], nodeConfigMap)
cloud/pkg/devicecontroller/controller/downstream.go:576:                        if isNodeSelectorUpdated(cachedDevice.Spec.NodeSelector, device.Spec.NodeSelector) {
cloud/pkg/devicecontroller/controller/downstream.go:599:                                        resource, err := messagelayer.BuildResource(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0], "device/"+device.Name+"/twin/cloud_updated", "")
cloud/pkg/devicecontroller/controller/downstream.go:697:        if len(device.Spec.NodeSelector.NodeSelectorTerms) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values) != 0 {
cloud/pkg/devicecontroller/controller/downstream.go:698:                configMap, ok := dc.configMapManager.ConfigMap.Load(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0])
cloud/pkg/devicecontroller/controller/downstream.go:715:                        dc.configMapManager.ConfigMap.Delete(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0])
cloud/pkg/devicecontroller/controller/downstream.go:720:                dc.configMapManager.ConfigMap.Store(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0], nodeConfigMap)
cloud/pkg/devicecontroller/controller/downstream.go:811:        if len(device.Spec.NodeSelector.NodeSelectorTerms) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values) != 0 {
cloud/pkg/devicecontroller/controller/downstream.go:812:                resource, err := messagelayer.BuildResource(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0], "membership", "")
cloud/pkg/synccontroller/createfailedobject.go:94:              nodeName := device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0]
```
**Environment**:
- KubeEdge version(e.g. `cloudcore/edgecore --version`):  
v.1.3.0

**CloudSide Environment**:
- Hardware configuration (e.g. `lscpu`):
- OS (e.g. `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Go version (e.g. `go version`): 
- Others:

**EdgeSide Environment**:
- edgecore version (e.g. `edgecore --version`):
- Hardware configuration (e.g. `lscpu`):
- OS (e.g. `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Go version (e.g. `go version`): 
- Others:

--- a/build/crds/devices/devices_v1alpha1_device.yaml
+++ b/build/crds/devices/devices_v1alpha1_device.yaml
@@ -167,6 +167,7 @@ spec:
               type: object
           required:
             - deviceModelRef
+            - nodeSelector
           type: object
         status:
           properties:

